v2/devdoc/log_context_property_basic_types_requirements.md

```diff
@@ -82,4 +82,339 @@ typedef LOG_CONTEXT_PROPERTY_TYPE (*LOG_CONTEXT_PROPERTY_TYPE_GET_TYPE)(void);
 
 **SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_010: [** `LOG_CONTEXT_PROPERTY_TYPE_IMPL(int64_t).get_type` shall returns the property type `LOG_CONTEXT_PROPERTY_TYPE_int64_t`. **]**
 
-Note: the rest of the types will be implemented by cloning int32_t, thus will be done in a separate PR.
+## LOG_CONTEXT_PROPERTY_TYPE_IMPL(uint64_t).to_string
+
+```c
+typedef int (*LOG_CONTEXT_PROPERTY_TYPE_TO_STRING)(void* property_value, char* buffer, size_t buffer_length);
+```
+
+`LOG_CONTEXT_PROPERTY_TYPE_IMPL(uint64_t).to_string` produces the string representation of an `uint64_t`.
+
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_011: [** If `property_value` is `NULL`, `LOG_CONTEXT_PROPERTY_TYPE_IMPL(uint64_t).to_string` shall fail and return -1. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_012: [** `LOG_CONTEXT_PROPERTY_TYPE_IMPL(uint64_t).to_string` shall call `snprintf` with `buffer`, `buffer_length` and format string `PRIu64` and pass in the values list the `uint64_t` value pointed to be `property_value`. **]**
```

Comment:
>snprintf [](http://example.com/codeflow?start=116&length=8)

One does have to wonder how much time does this logging take in our SW stack. And since we log pretty much everything, it miiiight be interesting to see if using something else except snprintf might be faster. 

Just thinking that snprintf can handle eeeeeeverything and we actually know beforehand the nature of the data to be stringified and the destination... maybe something else could be used for this purpose if performance is better with `stroull` for example, or even a hand crafted function.

Reply:
Agree, but let's not optimize it ... yet. the ETW sink will actually not stringify these, so we'd optimize the console logs only. Let's create a task for it.